packages/client/src/runtime/utils/CallSite.ts

```diff
@@ -0,0 +1,64 @@
+import * as stackTraceParser from 'stacktrace-parser'
+
+import { ErrorFormat } from '../getPrismaClient'
+
+export type LocationInFile = {
+  fileName: string
+  lineNumber: number | null
+  columnNumber: number | null
+}
+
+export interface CallSite {
+  getLocation(): LocationInFile | null
+}
+
+class DisabledCallSite implements CallSite {
+  getLocation(): LocationInFile | null {
+    return null
+  }
+}
+
+class EnabledCallSite implements CallSite {
+  private _error: Error
+  constructor() {
+    this._error = new Error()
+  }
+  getLocation(): LocationInFile | null {
+    const stack = this._error.stack
+    if (!stack) {
+      return null
+    }
+    const stackFrames = stackTraceParser.parse(stack)
+    // TODO: more resilient logic to check that it's not relative to cwd
+    const frame = stackFrames.find((t) => {
+      // Here we are trying to find the location in the users code which caused the error
+      return (
+        t.file &&
+        t.file !== '<anonymous>' && // Ignore as we can not read an <anonymous> file
+        !t.file.includes('@prisma') &&
+        !t.file.includes('getPrismaClient') &&
+        !t.file.startsWith('internal/') && // We don't want internal nodejs files
+        !t.methodName.includes('new ') &&
+        !t.methodName.includes('getCallSite') &&
+        !t.methodName.includes('Proxy.') &&
+        t.methodName.split('.').length < 4
```

Comment:
This could use a few more comments (if we actually understand why)

Reply:
I could not trigger about a half of those back then, but at the same time, not 100% confident we could remove them.